packages/fast-components-react-base/src/slider/slider.tsx

```diff
@@ -894,6 +908,99 @@ class Slider extends Foundation<SliderHandledProps, SliderUnhandledProps, Slider
     };
 
+    /**
+     * Handles touch dragging
+     */
+    private handleUpperThumbTouchStart = (e: React.TouchEvent): void => {
+        this.handleThumbTouchStart(e, SliderThumb.upperThumb);
+    };
+
+    private handleLowerThumbMouseTouchStart = (e: React.TouchEvent): void => {
+        this.handleThumbTouchStart(e, SliderThumb.lowerThumb);
+    };
+
+    private handleThumbTouchStart = (e: React.TouchEvent, thumb: SliderThumb): void => {
+        if (e.defaultPrevented || this.isBusyOrDisabled()) {
+            return;
+        }
+        
+        e.preventDefault();
+        (e.target as HTMLElement).focus();
+        window.addEventListener("touchend", this.handleTouchEnd);
+        window.addEventListener("touchcancel", this.handleTouchEnd);
+        window.addEventListener("touchmove", this.handleTouchMove);
+        this.setState({
+            isTouchDragging: true,
+            activeThumb: thumb,
+        });
+        const thisTouch: Touch = e.nativeEvent.touches.item(0);
+        this.updateDragValue(this.getTouchDragValue(thisTouch, thumb), thumb);
+    };
+
+    /**
+     *  Returns first valid touch found in a touch event
+     */
+    private getValidTouch = (event: TouchEvent): Touch => {
+        if (isNil(this.rootElement.current)) {
+            return null;
+        }
+        for (let i: number = 0; i < event.touches.length; i++) {
+            const thisTouch: Touch = event.touches.item(i);
+            const touchElement: HTMLElement = thisTouch.target as HTMLElement;
+            if (
+                touchElement.attributes[Slider.rolePropName].value === "slider" &&
+                touchElement.tabIndex === 0 &&
+                this.rootElement.current.contains(touchElement)
+            ) {
+                return thisTouch;
+            }
+        }
+        return null;
+    };
+
+    /**
+     *  Handle mouse moves during a thumb drag operation
+     */
+    private handleTouchMove = (event: TouchEvent): void => {
+        if (this.props.disabled || event.defaultPrevented) {
+            return;
+        }
+        const validTouch: Touch = this.getValidTouch(event);
+        if (validTouch === null) {
+            return;
+        }
+
+        this.updateDragValue(
+            this.getTouchDragValue(validTouch, this.state.activeThumb),
+            this.state.activeThumb
+        );
+    };
+
+    /**
+     *  Get dragvalue from touch event
+     */
+    private getTouchDragValue = (touch: Touch, thumb: SliderThumb): number => {
+        this.updateDirection();
+        this.updateSliderDimensions();
+        const pixelCoordinate: number =
+            this.props.orientation === SliderOrientation.vertical
+                ? touch.clientY
+                : touch.clientX;
+        const dragValue: number =
+            (this.props.range.maxValue - this.props.range.minValue) *
+                this.convertPixelToPercent(pixelCoordinate) +
+            this.props.range.minValue;
+        return dragValue;
```

Comment:
Seems like this method copies a lot of logic from `handleMouseMove`, can we consolidate it a bit and get the number of added lines down?

Reply:
Yes, consolidated into one fn that takes either a touch or mouseevent as a param